torch/distributed/_tensor/_dispatch.py

```diff
@@ -231,6 +237,19 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
+                if op_call == aten._amp_foreach_non_finite_check_and_unscale_.default:
+                    grad_placements = cast(list[dtensor.DTensor], args[0])[0].placements
+                    found_inf_placements: list[Placement] = []
+                    for placement in grad_placements:
+                        if isinstance(placement, Replicate):
+                            found_inf_placements.append(placement)
+                        else:
+                            found_inf_placements.append(Partial("max"))
+                    found_inf_dtensor = dtensor.DTensor.from_local(
```

Comment:
We should not call `from_local` inside `_torch_dispatch_`, but rather we should directly use `DTensor(..)` here

Reply:
Replaced by DTensor(..) now